internal/impl/postgresql/pglogicalstream/connection.go

```diff
@@ -9,20 +9,29 @@
 package pglogicalstream
 
 import (
+	"crypto/tls"
 	"database/sql"
 	"fmt"
 	"regexp"
 	"strconv"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/jackc/pgx/v5/stdlib"
 )
 
 var re = regexp.MustCompile(`^(\d+)`)
 
-func openPgConnectionFromConfig(dbDSN string) (*sql.DB, error) {
-	return sql.Open("postgres", dbDSN)
+func openPgConnectionFromConfig(dbDSN string, tlsConfig *tls.Config) (*sql.DB, error) {
+	config, err := pgxpool.ParseConfig(dbDSN)
+	if err != nil {
+		return nil, err
+	}
+	config.ConnConfig.TLSConfig = tlsConfig
+	return stdlib.OpenDB(*config.ConnConfig), nil
 }
 
 func getPostgresVersion(dbDSN string) (int, error) {
-	conn, err := openPgConnectionFromConfig(dbDSN)
+	conn, err := openPgConnectionFromConfig(dbDSN, nil)
```

Comment:
This is required to pass in the TLSConfig as well otherwise the version checks will fail